x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engine/engines_lens/engine_analytics_lens_logic.ts

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { kea, MakeLogicType } from 'kea';
+
+import { DataView } from '@kbn/data-views-plugin/public';
+import { FormulaPublicApi, TypedLensByValueInput } from '@kbn/lens-plugin/public';
+
+import { getLensXYLensAttributes, getMetricLensAttributes } from './lens_data';
+
+interface LensArguments {
+  defaultDataView: DataView | null;
+  formula: FormulaPublicApi;
+  isNoResultsCardVisible?: boolean;
+}
+type LensResponse = TypedLensByValueInput['attributes'];
+
+interface EngineAnalyticsLensValues {
+  metricAttributesNoResults: LensResponse;
+  metricAttributesQueries: LensResponse;
+  resultMetricAttributesNoResults: LensResponse | null;
+  resultMetricAttributesQueries: LensResponse | null;
+  resultXYAttributes: LensResponse | null;
+  xyAttributes: LensResponse;
+}
+interface EngineAnalyticsLensActions {
+  getLensAreaAttributes: (args: LensArguments) => LensArguments;
+  getLensMetricAttributes: (args: LensArguments) => LensArguments;
+  setLensAreaAttributes: (resultXYAttributes: EngineAnalyticsLensValues['resultXYAttributes']) => {
+    resultXYAttributes: EngineAnalyticsLensValues['resultXYAttributes'];
+  };
+  setLensMetricAttributesNoResults: (
+    resultMetricAttributesNoResults: EngineAnalyticsLensValues['resultMetricAttributesNoResults']
+  ) => {
+    resultMetricAttributesNoResults: EngineAnalyticsLensValues['resultMetricAttributesNoResults'];
+  };
+  setLensMetricAttributesQueries: (
+    resultMetricAttributesQueries: EngineAnalyticsLensValues['resultMetricAttributesQueries']
+  ) => {
+    resultMetricAttributesQueries: EngineAnalyticsLensValues['resultMetricAttributesQueries'];
+  };
+}
+
+export const EngineAnalyticsLensLogic = kea<
+  MakeLogicType<EngineAnalyticsLensValues, EngineAnalyticsLensActions>
+>({
+  actions: {
+    getLensAreaAttributes: (args: LensArguments) => args,
+    getLensMetricAttributes: (args: LensArguments) => args,
+    setLensAreaAttributes: (resultXYAttributes: LensResponse) => ({
+      resultXYAttributes,
+    }),
+    setLensMetricAttributesNoResults: (resultMetricAttributesNoResults: LensResponse) => ({
+      resultMetricAttributesNoResults,
+    }),
+    setLensMetricAttributesQueries: (resultMetricAttributesQueries: LensResponse) => ({
+      resultMetricAttributesQueries,
+    }),
+  },
+
+  listeners: ({ actions }) => ({
+    getLensAreaAttributes: (input) => {
```

Comment:
suggestion: in general, you can use destructuring to make this a little nicer. Eg:
```
getLensAreaAttributes: ({ defaultDataView, formula, isNoResultsCardVisible }) => {
  actions.setLensAreaAttributes(defaultDataView, formula, isNoResultsCardVisible)
}
```